### ISSUE TITLE
Clarifying documentation build instructions

### ISSUE THREAD

**Issue by mrl280** [qoherent/ria#7]
Thank you for submitting a feature request. We value your feedback. Before proceeding with a new issue, please 
take a moment to review the issue tracker for any existing requests related to the same feature.

Please be aware that RIA Core serves as the open-source core of a broader software engineering initiative. 
The feature you're requesting might already be available within [RIA Hub](https://riahub.ai/) or through an 
alternative Qoherent solution. In such cases, we will, during our triaging process, provide any relevant 
information in the issue comments and label the request with `wont fix ❌`.

**Description:**
Continuing the discussion from #6 , regarding the documentation build instructions in CODING.md.

**Project Area:**
To which aspect of RIA Core does this request pertain?
- [ ] The enhancement or addition of an importable module (i.e., using `import ria` in your Python project) 
- [ ] The enhancement or addition or a project script (i.e., executing `ria curate` from the command line)
- [ ] Our radio interface tools, such as requesting support for a new radio
- [X] Project documentation or tutorials, including new or improved code snippets and graphics
- [ ] CI and DevOps, such as new or improved unit tests or GitHub workflows

**Additional context:**

> Instructions for building docs with `sphinx-autobuild` are all good. One thing I want to clarify further is the types of changes that get built automatically. I think explicitly mentioning that only changes within the `source` directory, including the `.rst` files and `_templates/layout.html` will get built automatically will probably be useful. Let me know your thoughts!

Thanks @UmairK5669. Ya, when we run `sphinx-build`, it will process the files in the `source` directory. When processing the `automodule` directives, Sphinx should generate the corresponding HTML files from the project source. So, changes to docstrings should automatically be captured in this way. 

As I understand, the only thing that is not automatic is adding the `automodule` directives for new modules, for which I use `sphinx-apidoc`, by running `sphinx-apidoc -o ./source ../ria` or similiar, and then you have to manually update the links in the `index.rst` files. Sounds like lots of work has been done to automate this, so perhaps we should adopt one of these work arounds: https://github.com/readthedocs/readthedocs.org/issues/1139? Also, I guess this also means that right now `sphinx-apidoc` is not technically a project dependency, but just a tool we suggest using. 

Perhaps for now the I should just update step 3 to read: "A working copy of the docs will now be available at http://localhost:8000/, and any changes made to the configurations in the `docs/source` directory will be automatically reflected there. Docstring changes will be automatically reflected for all components with `automodule` directives." And then further explain the use of `sphinx-apidoc` in the note at the bottom? Let me know if you think these changes are accurate and would strengthen the instructions, or if you had something else in mind.

_Originally posted by @mrl280 in https://github.com/qoherent/ria/issues/6#issuecomment-1989002991_

Once your feature request has been submitted, it will be triaged by an authorized Qoherent team member. 
Please wait for the issue to be triaged before starting development. This is to ensure your efforts are 
focused effectively. Thank you for your patience and understanding.

**Comment by UmairK5669** [comment]
> Perhaps for now the I should just update step 3 to read: "A working copy of the docs will now be available at http://localhost:8000/, and any changes made to the configurations in the docs/source directory will be automatically reflected there. Docstring changes will be automatically reflected for all components with automodule directives." And then further explain the use of sphinx-apidoc in the note at the bottom? Let me know if you think these changes are accurate and would strengthen the instructions, or if you had something else in mind.

These new instructions are great. I don't necessarily think we need to further explain sphinx-apidoc as you mention its use clearly above, however, if you do think further clarification is necessary/useful then it should be added.
